sdk/core/azure-core/inc/context.hpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <assert.h>
+#include <chrono>
+#include <memory>
+//#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <type_traits>
+
+constexpr char not_found = 0;
+
+namespace Azure
+{
+namespace Core
+{
+
+struct ValueBase
+{
+  virtual ~ValueBase() {}
+};
+
+/*
+ *  @brief ContextValue exists as a substitute for variant which isn't available until C++17
+ */
+class ContextValue
+{
+  char active;
```

Comment:
These are still lowercase.

Reply:
I think you should an enum with the possible values & their meanings instead of comments.
The code that check will be more self-documenting.